pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -0,0 +1,219 @@
+//go:build linux
+// +build linux
+
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	"os/exec"
+	"testing"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/types"
+	utilstest "github.com/kinvolk/inspektor-gadget/pkg/gadgets/utils/test"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+	"golang.org/x/sys/unix"
+)
+
+func TestCreateTracer(t *testing.T) {
+	if unix.Getuid() != 0 {
+		t.Skip("test requires root")
+	}
+
+	eventCallback := func(types.Event) {}
+
+	tracer, err := NewTracer(&Config{}, nil, eventCallback)
+	if err != nil {
+		t.Fatalf("error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("returned tracer was nil")
+	}
+	defer tracer.Stop()
+
+	// check that a double stop doesn't cause issues
+	tracer.Stop()
+	tracer.Stop()
+}
+
+func TestTracer(t *testing.T) {
+	if unix.Getuid() != 0 {
+		t.Skip("test requires root")
+	}
+
+	// variable to save the value of the created process PID
+	var catPid int
+
+	manyArgs := []string{}
+	// 19 is DEFAULT_MAXARGS - 1
+	for i := 0; i < 19; i++ {
+		manyArgs = append(manyArgs, "/dev/null")
+	}
+
+	// function to generate an event used most of the times
+	generateEvent := func() error {
+		cmd := exec.Command("/bin/cat", "/dev/null")
+		if err := cmd.Run(); err != nil {
+			return err
+		}
+
+		catPid = cmd.Process.Pid
+
+		return nil
+	}
+
+	testRunner := utilstest.NewTestRunner[types.Event](t)
+
+	tests := map[string]utilstest.Test[types.Event]{
+		"test without filtering": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				tracer, err := NewTracer(&Config{}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+			GetAnyExpectedEvent: func(info *utilstest.Info) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Pid:       uint32(catPid),
+					Ppid:      uint32(info.Pid),
+					UID:       uint32(info.UID),
+					MountNsID: info.MountNsID,
+					Retval:    0,
+					Comm:      "cat",
+					Args:      []string{"/bin/cat", "/dev/null"},
+				}
+			},
+		},
+		"test filtering (no event should be captured)": {
```

Comment:
Test names with spaces will be converted to something like `test_filtering_(no_event_should_be_captured)` by `go test`, which makes it harder to find by name.

Reply:
Ah yes, because you don't use `t.Run`.